google-cloud-examples/src/test/java/com/google/cloud/examples/pubsub/snippets/ITPubSubSnippets.java

```diff
@@ -264,4 +277,79 @@ public void testTopicPolicyAsync() throws ExecutionException, InterruptedExcepti
     topic.delete();
     subscription.delete();
   }
+
+  @Test
+  public void testPublisherSubscriber() throws Exception {
+    TopicName topicName =
+        TopicName.create(pubsub.getOptions().getProjectId(), formatForTest("test-topic"));
+    SubscriptionName subscriptionName =
+        SubscriptionName.create(
+            pubsub.getOptions().getProjectId(), formatForTest("test-subscription"));
+    try (PublisherClient publisherClient = PublisherClient.create();
+        SubscriberClient subscriberClient = SubscriberClient.create()) {
+      publisherClient.createTopic(topicName);
+      subscriberClient.createSubscription(
+          subscriptionName, topicName, PushConfig.getDefaultInstance(), 0);
+
+      testPublisherSubscriberHelper(topicName, subscriptionName);
+
+      subscriberClient.deleteSubscription(subscriptionName);
+      publisherClient.deleteTopic(topicName);
+    }
+  }
+
+  private void testPublisherSubscriberHelper(
+      TopicName topicName, SubscriptionName subscriptionName) throws Exception {
+    String messageToPublish = "my-message";
+
+    Publisher publisher = null;
+    try {
+      publisher = Publisher.newBuilder(topicName).build();
+      PublisherSnippets snippets = new PublisherSnippets(publisher);
+      final SettableRpcFuture<Void> done = new SettableRpcFuture<>();
+      snippets
+          .publish(messageToPublish)
+          .addCallback(
+              new RpcFutureCallback<String>() {
+                public void onSuccess(String messageId) {
+                  done.set(null);
+                }
+
+                public void onFailure(Throwable t) {
+                  done.setException(t);
+                }
+              });
+      done.get();
+    } finally {
+      if (publisher != null) {
+        publisher.shutdown();
+      }
+    }
+
+    final BlockingQueue<PubsubMessage> queue = new ArrayBlockingQueue<>(1);
+    final SettableRpcFuture<Void> done = new SettableRpcFuture<>();
+    final SettableRpcFuture<PubsubMessage> received = new SettableRpcFuture<>();
+    SubscriberSnippets snippets =
+        new SubscriberSnippets(
+            subscriptionName,
+            new MessageReceiverSnippets(queue).messageReceiver(),
+            done,
+            MoreExecutors.directExecutor());
+    new Thread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          received.set(queue.poll(10, TimeUnit.MINUTES));
```

Comment:
I agree that this is a long time to wait. On the other hand, if this times out, the whole test fails anyway right?